packages/frontend-plugin-api/src/types.ts

```diff
@@ -14,9 +14,37 @@
  * limitations under the License.
  */
 
+import { BackstagePlugin } from '@backstage/core-plugin-api';
+import { ComponentType, PropsWithChildren } from 'react';
+
 // TODO(Rugvip): This might be a quite useful utility type, maybe add to @backstage/types?
 /**
  * Utility type to expand type aliases into their equivalent type.
  * @ignore
  */
 export type Expand<T> = T extends infer O ? { [K in keyof O]: O[K] } : never;
+
+/** @public */
+export type CoreProgressComponent = ComponentType<PropsWithChildren<{}>>;
+
+/** @public */
+export type CoreBootErrorPageComponent = ComponentType<
+  PropsWithChildren<{
+    step: 'load-config' | 'load-chunk';
+    error: Error;
+  }>
+>;
+
+/** @public */
+export type CoreNotFoundErrorPageComponent = ComponentType<
+  PropsWithChildren<{}>
+>;
+
+/** @public */
+export type CoreErrorBoundaryFallbackComponent = ComponentType<
+  PropsWithChildren<{
+    plugin?: BackstagePlugin;
```

Comment:
The legacy backstage plugin is used here for backward compatibility. Should we use the new ExtensionBoundary as the type here and convert the plugins inside the legacy adapter function?

Reply:
Yeah I think using the new plugin here is the way to go, and then convert in the adapter. We might also want to switch it over to being an app node or something like that instead as that'd provide a bit more info, including the plugin if available

Reply:
Thanks, I will address this in a follow-up PR 🙂

Reply:
https://github.com/backstage/backstage/pull/21652